Remove pbrun's '-b' option.

qbit opened the pull request:
##### ISSUE TYPE
- Feature Pull Request
##### ANSIBLE VERSION

<!--- Paste verbatim output from “ansible --version” between quotes below -->

```
ansible 2.3.0
```
##### COMPONENT NAME

```
ansible
```
##### SUMMARY

This PR removes the '-b' option. It is still possible to pass it by setting `become_flags`.

I am not 100% sure why but `assword:` stopped working (Timeout waiting for esc prompt) with the rest of the changes.. but `Password:` works. Would anyone be able to shed some light on that? I left it as an extra commit so it can be cherry picked out if needed.

bcoca commented on lib/ansible/constants.py:
```diff
@@ -235,6 +235,7 @@ def load_config_file():
 DEFAULT_SUDO_EXE          = get_config(p, DEFAULTS, 'sudo_exe', 'ANSIBLE_SUDO_EXE', None)
 DEFAULT_SUDO_FLAGS        = get_config(p, DEFAULTS, 'sudo_flags', 'ANSIBLE_SUDO_FLAGS', '-H -S -n')
 DEFAULT_ASK_SUDO_PASS     = get_config(p, DEFAULTS, 'ask_sudo_pass',    'ANSIBLE_ASK_SUDO_PASS',    False, value_type='boolean')
+DEFAULT_PBRUN_FLAGS       = get_config(p, DEFAULTS, 'pbrun_flags', 'ANSIBLE_PBRUN_FLAGS', '-b')
```
we are moving away from 'per method flags' these should be in 'become flags'

qbit replied:
OK, I will remove this.

bcoca commented on lib/ansible/playbook/play_context.py:
```diff
@@ -517,8 +518,8 @@ def detect_su_prompt(data):
 
-                prompt='assword:'
-                becomecmd = '%s -b %s -u %s %s' % (exe, flags, self.become_user, success_cmd)
+                prompt='Password:'
```
-b should appear if no password is supplied, we can make conditional like with sudo, but it is there to avoid prompts, when ansible is not expecting them

qbit replied:
Sounds good.

bcoca commented:
not sure about the differences with the password prompt, since i don't have access to pbrun to test I have to rely on user and contributor feedback.

qbit commented:
I am getting a test instance together so I can test -b changes.. I will also update the diff here shortly.

qbit committed:
Remove -b option from pbrun.

The -b option reads as follows:
` The target job is directed to ignore hangup signals. This is particularly
useful for running the target program in the background.`

If needed, '-b' can be added to become_flags

Squashed commit of the following:

commit f2c9f5c011ae8be610301d597a34bfba1a391e08
Author: Aaron Bieber <aaron@bolddaemon.com>
Date:   Mon Oct 17 10:58:14 2016 -0600

    remove pbrun flags

commit f402679ac177c931ad64bd13306f62512a14fcd6
Author: Aaron Bieber <aaron@bolddaemon.com>
Date:   Fri Oct 14 15:29:29 2016 -0600

    use Password: vs assword: for matching pbrun prompt

commit cd2e90cb65854c4cc5dd8773404e520d40f82765
Author: Aaron Bieber <aaron@bolddaemon.com>
Date:   Fri Oct 14 15:28:58 2016 -0600

    move -b to pbrun_flags

qbit commented:
@bcoca updated.
